### PR TITLE
Change recovery list colors

### PR DESCRIPTION
**Pull request by nmattia**
This updates the colors used to show the recovery words and adds some extra machinery in the styleguide for making words editable.

In particular:

* The gradient is removed from the recovery list as it was confusing to people (it wasn't clear why some words were green and some red)
* A couple classes are added to support "confirming" the phrase. In particular we need to show some words as requiring attention, being incorrectly spelled, etc.

https://user-images.githubusercontent.com/6930756/224682895-5061c0da-1835-4c29-a146-8a8bfc9a187e.mov

### CODE REVIEW

**Review by meodai — APPROVED** [review]

File: src/frontend/src/styles/main.css [hunk 1]
```diff
@@ -1354,22 +1369,94 @@ a:hover,
...
+}
+
+.c-list--recovery-word__incorrect::before {
+  color: var(--bg);
+}
+
+.c-recoveryInput {
+  display: block;
+  width: 100%;
+  text-align: left;
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:1453]: redundant

File: src/frontend/src/styles/main.css [hunk 2]
```diff
@@ -1354,22 +1369,94 @@ a:hover,
...
+  --bg: var(--rc-button);
+}
+
+/* When a word requires attention, show it as green and show a checkmark instead of the counter */
+.c-list--recovery-word__correct {
+  --bg: var(--rc-input-success);
+}
+
+.c-list--recovery-word__correct::before {
+  color: var(--bg);
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:1433]: used to hide counter; note: unexpected

File: src/frontend/src/styles/main.css [hunk 3]
```diff
@@ -1354,22 +1369,94 @@ a:hover,
...
+  margin-left: -0.05em;
+}
+
+/* When a word requires attention, show it as red and show a warning sign instead of the counter */
+.c-list--recovery-word__incorrect {
+  --bg: var(--rc-input-error);
+}
+
+.c-list--recovery-word__incorrect::before {
+  color: var(--bg);
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:1447]: used to hide counter; note: unexpected

File: src/frontend/src/styles/main.css [hunk 4]
```diff
@@ -1297,53 +1308,57 @@ a:hover,
...
   font-family: monospace;
   text-align: center;
 
   z-index: 1;
 
-  --current-color: hsl(calc(150 + var(--i) * -180), 100%, 93%);
+  /* The counter displayed. We set this explicitly from the HTML because
+   * the index shown is important for recovery. Also this works around issues related
+   * to different counter behavior on Safari & Chrome. */
+  --counter: var(--index);
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:1344]: Redundant and now incorrect

File: src/frontend/src/styleguide.ts [hunk 5]
```diff
@@ -320,16 +322,57 @@ export const styleguide = html`
...
+                        >${icon === "warning" ? warningIcon : checkmarkIcon}</i
+                      >`
+                    : null}
+                  ${contenteditable === true
+                    ? html`<input
+                          type="text"
+                          class="c-recoveryInput"
+                          value=${text}
+                          pattern=${text}
+                          required
```
Comment (nmattia) [inline src/frontend/src/styleguide.ts:370]: Unused; validation is done in js

File: src/frontend/src/styleguide.ts [hunk 6]
```diff
@@ -320,16 +322,57 @@ export const styleguide = html`
...
+                          maxlength="8"
+                        />&nbsp;`
```
Comment (nmattia) [inline src/frontend/src/styleguide.ts:372]: Unexpected

File: src/frontend/src/flows/recovery/confirmSeedPhrase.ts [hunk 7]
```diff
@@ -105,25 +105,56 @@ const confirmSeedPhraseTemplate = ({
 export const wordTemplate = ({
   word,
   update,
+  i,
 }: {
   word: Word;
   /* Notify the caller that a word was updated */
   update: () => void;
+  i: number;
```
Comment (nmattia) [inline src/frontend/src/flows/recovery/confirmSeedPhrase.ts:113]: naming should be clearer and functionality clarified

File: src/frontend/src/flows/recovery/displaySeedPhrase.ts [hunk 8]
```diff
@@ -51,12 +51,16 @@ const displaySeedPhraseTemplate = ({
...
+        <output class="c-input c-input--recovery"
           ><ol translate="no" class="c-list c-list--recovery">
             ${recoveryWords.map(
               (word, i) =>
-                html`<li style="--i: ${i / recoveryWords.length}">${word}</li>`
+                html`<li
+                  class="c-list--recovery-word"
+                  style="--index: '${i + 1}';"
+                >
+                  <i>${word}</i>
```
Comment (nmattia) [inline src/frontend/src/flows/recovery/displaySeedPhrase.ts:62]: `<i>` wrapper is unnecessary

File: src/frontend/src/styleguide.ts [hunk 9]
```diff
@@ -320,16 +322,57 @@ export const styleguide = html`
...
-          >
+          <output class="c-input c-input--recovery">
             <ol class="c-list c-list--recovery">
-              ${recoveryWords.map(
-                (word, i) =>
-                  html`<li style="--i: ${i / recoveryWords.length}">
-                    ${word}
-                  </li>`
-              )}
+              ${recoveryWords.map((word, i) => {
```
Comment (nmattia) [inline src/frontend/src/styleguide.ts:327]: unclear; should explain what's happening here

File: src/frontend/src/styleguide.ts [hunk 10]
```diff
@@ -320,16 +322,57 @@ export const styleguide = html`
...
+                const classes = ["c-list--recovery-word"];
+                let contenteditable: undefined | true;
```
Comment (nmattia) [inline src/frontend/src/styleguide.ts:329]: name is off, and should just be a boolean

File: src/frontend/src/styles/main.css [hunk 11]
```diff
@@ -1354,22 +1369,94 @@ a:hover,
...
+.c-list--recovery-word__correct::before,
+.c-list--recovery-word__correct .c-list--recovery-word__icon {
+  --fg: var(--rc-onButton);
+  color: var(--fg);
+}
+
+/* When a word is disabled, bring the saturation down to make it look non-interactive */
+.c-list--recovery-word__disabled {
+  --bg: var(--rc-input-readonly);
+  --fg: var(--rc-onInput-readonly);
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:1419]: should probably explain what bg and fg are for

**Pull request merged**